### PR TITLE
Docs: Fedramp updates

### LINKED ISSUE #460: Update FedRAMP docs page
## Applies To

https://goteleport.com/docs/access-controls/compliance-frameworks/fedramp/

## Details

We need to update the list of controls sourced from the webpage 
https://goteleport.com/use-cases/fedramp-compliance/

### PR DESCRIPTION
**Pull request by mmcallister**
Added supported FedRAMP control features.
Updated the new base in `config.json` for the **SP 800-53** set of controls. We'll have to periodically update it there; the versioning changes over time but the way they have the site map organized, the other option would be to use their base page here:

https://csrc.nist.gov/projects/cprt/catalog#/cprt/home

and then customers can look for the latest revision for `SP 800-53 Rev x.x.x` but that's tedious and could be confusing for folks.

### CODE REVIEW

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 1]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
+| [AC-03 Access Enforcement]((=fedramp.control_url=)AC-03) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which SSH nodes a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |
+| [AC-07 Unsuccessful Logon Attempts]((=fedramp.control_url=)AC-07) |  Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which SSH nodes a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |	Teleport supports two types of users: local and SSO-based accounts (GitHub, Google Apps, Okta, etc). For local accounts, by default, Teleport locks accounts for 30 minutes after 5 failed login attempts. For SSO-based accounts, the number of invalid login attempts and lockout time period is controlled by the SSO provider. |
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:22]:
```suggestion
| [AC-07 Unsuccessful Logon Attempts]((=fedramp.control_url=)AC-07) |  Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which resources a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |	Teleport supports two types of users: local and SSO-based accounts (GitHub, Google Apps, Okta, etc). For local accounts, by default, Teleport locks accounts for 30 minutes after 5 failed login attempts. For SSO-based accounts, the number of invalid login attempts and lockout time period is controlled by the SSO provider. |
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 2]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
+| [AC-08 System Use Notification]((=fedramp.control_url=)AC-08) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which SSH nodes a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |	Teleport integrates with Linux Pluggable Authentication Modules (PAM). PAM modules can be used to display a custom message on login using a message of the day (MOTD) module within the Session management primitive. |
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:23]:
```suggestion
| [AC-08 System Use Notification]((=fedramp.control_url=)AC-08) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which resources a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |	Teleport integrates with Linux Pluggable Authentication Modules (PAM). PAM modules can be used to display a custom message on login using a message of the day (MOTD) module within the Session management primitive. |
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 1 above)
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:22]: Looks like "Unix" is slightly more popular than "UNIX" in our docs, but I don't feel strongly either way.

```
teleport/docs on probakowski/keyboard-layout [$]
➜ ag --count Unix
pages/machine-id/reference/configuration.mdx:3
pages/includes/config-reference/ssh-service.yaml:1
pages/includes/metrics.mdx:2
pages/management/admin/labels.mdx:1
pages/management/admin/trustedclusters.mdx:1
pages/server-access/guides/host-user-creation.mdx:2
pages/server-access/guides/ssh-pam.mdx:7
pages/architecture/authorization.mdx:1
pages/reference/cli/tctl.mdx:6

teleport/docs on probakowski/keyboard-layout [$]
➜ ag --count UNIX
pages/access-controls/compliance-frameworks/fedramp.mdx:1
pages/access-controls/compliance-frameworks/soc2.mdx:1
pages/access-controls/guides/locking.mdx:1
pages/access-controls/sso.mdx:1
pages/includes/role-spec.mdx:1
pages/server-access/rbac.mdx:1
pages/architecture/authorization.mdx:1
pages/installation.mdx:1
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 3]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
-[Teleport account](https://teleport.sh). Look for `Linux 64-bit (FedRAMP/FIPS)`.
+| Control | Teleport Features |
+| - | - |
+| [CM-08 Information System Component Inventory]((=fedramp.control_url=)CM-08) | Teleport maintains a live list of all nodes within a cluster. This node list can be queried by users (who see a subset they have access to) and administrators any time. |
+
+### Identification and Authentication
 
+| Control | Teleport Features |
+| - | - |
+| [IA-02 Concurrent Session Control]((=fedramp.control_url=)IA-02) | Integrates with SSO providers such as GitHub, Okta, Google, etc. Acts as its own SSO provider. Enforces the use of multi-factor authentication (MFA), including requiring per-session MFA. Supports PIV-compatible hardware keys. |
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:46]: We also support `max_connections` and `max_users`: https://goteleport.com/docs/management/operations/scaling/#auth-service-and-proxy-service-configuration

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 4]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
+
+### System and Communications Protection
+
+| Control | Teleport Features |
+| - | - |
+| [SC-10 Network Disconnection]((=fedramp.control_url=)SC-10) | Teleport requires valid X.509 or SSH certificates issued by a Teleport Certificate Authority (CA) to establish a network connection for device-to-device network connection between Teleport components. |
+| [SC-12 Cryptographic Key Establish and Management]((=fedramp.control_url=)SC-12) | Teleport initializes cryptographic keys that act as a Certificate Authority (CA) to further issue x509 and SSH certificates. SSH and x509 user certificates that are issued are signed by the CA and are (by default) short-lived. SSH host certificates are also signed by the CA and rotated automatically (a manual force rotation can also be performed).<br/>Teleport Enterprise builds against a FIPS 140-2 compliant library (BoringCrypto) is available. <br/>In addition, when Teleport Enterprise is in FedRAMP/FIPS 140-2 mode, Teleport will only start and use FIPS 140-2 compliant cryptography. |
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:56]: I don't think SSH host certificates are rotated automatically.

Should we mention HSM support here too?

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 5]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
 Teleport includes FedRAMP and FIPS 140-2 features to support companies that sell into
 government agencies.
 
-| Control | Teleport Features |
+### Access Controls
```
Comment (ptgott) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:16]:
```suggestion
### Access controls
```
Headings in the docs use sentence case: https://github.com/gravitational/docs/blob/main/docs-contributors/style-guide.md#headings

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 6]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
 | - | - |
-| [AC-02 Account Management]((=fedramp.control_url=)AC-2) | Audit events are emitted in the Auth Service when a user is created, updated, deleted, locked, or unlocked. |
-| [AC-03 Access Enforcement]((=fedramp.control_url=)AC-3) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which SSH nodes a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |
+| [AC-02 Account Management]((=fedramp.control_url=)AC-02) | Audit events are emitted in the Auth Service when a user is created, updated, deleted, locked, or unlocked. |
+| [AC-03 Access Enforcement]((=fedramp.control_url=)AC-03) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which SSH nodes a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |
```
Comment (ptgott) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:21]:
```suggestion
| [AC-03 Access Enforcement]((=fedramp.control_url=)AC-03) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which infrastructure resources a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 2 above)
Comment (ptgott) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:23]:
```suggestion
| [AC-08 System Use Notification]((=fedramp.control_url=)AC-08) | Teleport Enterprise supports robust [Role-based Access Controls (RBAC)](../../access-controls/introduction.mdx) to: <br/>• Control which infrastructure resources a user can or cannot access. <br/>• Control cluster level configuration (session recording, configuration, etc.) <br/>• Control which UNIX logins a user is allowed to use when logging into a server. |	Teleport integrates with Linux Pluggable Authentication Modules (PAM). PAM modules can be used to display a custom message on login using a message of the day (MOTD) module within the Session management primitive. |
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 7]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
-| [IA-03 Device Identification and Authentication]((=fedramp.control_url=)IA-3) | Teleport requires valid x509 or SSH certificates issued by a Teleport Certificate Authority (CA) to establish a network connection for device-to-device network connection between Teleport components. |
-| [SC-12 Cryptographic Key Establish and Management]((=fedramp.control_url=)SC-12) | Teleport initializes cryptographic keys that act as a Certificate Authority (CA) to further issue x509 and SSH certificates. SSH and x509 user certificates that are issued are signed by the CA and are (by default) short-lived. SSH host certificates are also signed by the CA and rotated automatically (a manual force rotation can also be performed).<br/>Teleport Enterprise builds against a FIPS 140-2 compliant library (BoringCrypto) is available. <br/>In addition, when Teleport Enterprise is in FedRAMP/FIPS 140-2 mode, Teleport will only start and use FIPS 140-2 compliant cryptography. |
 
-## Download and install
+### Configuration Management
```
Comment (ptgott) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:36]: This and other headings should also use sentence case

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 8]
```diff
@@ -13,32 +13,63 @@ Teleport FIPS mode works and how it can help your company to become FedRAMP auth
...
 
-Teleport Enterprise customers can download the custom FIPS package from their
-[Teleport account](https://teleport.sh). Look for `Linux 64-bit (FedRAMP/FIPS)`.
+| Control | Teleport Features |
+| - | - |
+| [CM-08 Information System Component Inventory]((=fedramp.control_url=)CM-08) | Teleport maintains a live list of all nodes within a cluster. This node list can be queried by users (who see a subset they have access to) and administrators any time. |
```
Comment (ptgott) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:40]: I think this guide predates the presence of multiple Teleport agent services (beyond the SSH Service), so it says "nodes" or "SSH nodes" where it should say something like "infrastructure resources".

**Review by ptgott — APPROVED** [review]

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 2 above)
Comment (mmcallister) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx:23]: Updated

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 1 above)
Comment (mmcallister) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx:22]: Updated for consistency

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 8 above)
Comment (mmcallister) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx:40]: ✔️

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 3 above)
Comment (mmcallister) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx:46]: It looks like user and connection limits aren't mandatory for the IA-02 control, but we could mention they're supported here. Thoughts, @ptgott ?

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 4 above)
Comment (mmcallister) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx:56]: HSM isn't part of SC-12 but it seems like a good thing to mention here. For the automatic cert rotation note, the same line is also in SC-17. From what I've read, FedRAMP doesn't mandate any sort of automatic certificate rotation, but from this it looks like we do support it, unless I am misunderstanding (under rotation types):
https://goteleport.com/docs/management/operations/ca-rotation/

> Semi-automatic: Teleport automatically transitions between phases of the rotation after some amount of time (known as a grace period) elapses.

(Note, if it does need to be removed we will need to update the source webpage as well: https://goteleport.com/use-cases/fedramp-compliance).

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 3 above)
Comment (ptgott) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx:46]: That sounds good to me!

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx (hunk 4 above)
Comment (mmcallister) [reply docs/pages/access-controls/compliance-frameworks/fedramp.mdx]: - Added HSM sentence. 
- Deleted rotation lines as discussed.

**Comment by mmcallister** [comment]
@zmb3 Would you mind taking a look at the changes? I need 1 more approval for this one. Thank you!

[Vercel preview](https://docs-rj08j2gya-goteleport.vercel.app/docs/)

**Review by ptgott — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**